internal/provider/kubernetes/controller.go

```diff
@@ -290,15 +290,23 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, request reconcile.
 						r.log.Info("processing Secret", "namespace", secretNamespace, "name", string(certRef.Name))
 
 						if secretNamespace != gtw.Namespace {
-							from := ObjectKindNamespacedName{kind: gatewayapi.KindGateway, namespace: gtw.Namespace, name: gtw.Name}
-							to := ObjectKindNamespacedName{kind: gatewayapi.KindSecret, namespace: secretNamespace, name: string(certRef.Name)}
+							from := ObjectKindNamespacedName{
+								kind:      gatewayapi.KindGateway,
+								namespace: gtw.Namespace,
+								name:      gtw.Name,
+							}
+							to := ObjectKindNamespacedName{
+								kind:      gatewayapi.KindSecret,
+								namespace: secretNamespace,
+								name:      string(certRef.Name),
+							}
 							refGrant, err := r.findReferenceGrant(ctx, from, to)
 							if err != nil {
-								r.log.Error(err, "unable to find ReferenceGrant that links the Secret to Gateway")
-								continue
+								return reconcile.Result{}, err
```

Comment:
this doesnt look right, we are returning early here, skipping processing and sending other correct info

Reply:
@arkodg this error is due to the client not being able to list ReferenceGrants. Do we want to recompute the resource tree under this circumstance and potentially cause routes to be inadvertently CRUD'd?

Reply:
this error looks like its for a specific refgrant object
we have two options here 
1) do nothing - which will result in user not being able to apply any new intent
2) skip processing bad object but process the rest - which will result in translating all readable intent
2 seems more correct as well as intuitive than 1

Reply:
Agreed that #2 is more intuitive than #1. I'll update the PR.

Reply:
thanks, I think if the `return` statement is removed, it should satisfy 1)

Reply:
Resolved by commit 753e8c7.